src/typeahead/typeahead.ts

```diff
@@ -206,7 +209,12 @@ export class NgbTypeahead implements ControlValueAccessor,
 
   registerOnTouched(fn: () => any): void { this._onTouched = fn; }
 
-  writeValue(value) { this._writeInputValue(this._formatItemForInput(value)); }
+  writeValue(value) {
+    this._writeInputValue(this._formatItemForInput(value));
+    if (this.showHint) {
```

Comment:
I don't think it is necessary, because the first thing you do on user input is already `this._inputValueBackup = value;`

Reply:
I removed it.